File: Comms Protocol Java/Comms Protocol Java/comms/protocol/java/Gyro_RawDataPacket.java
package comms.protocol.java;

public class Gyro_RawDataPacket extends AndroidDataPacket {
	private static int numberOfValues = 3;

	public Gyro_RawDataPacket() {
		super(ValidPacketTypes.gyro_raw_data_packet, numberOfValues);
	}

	public Gyro_RawDataPacket(byte[] payload) throws Exception {
		super(ValidPacketTypes.gyro_raw_data_packet, numberOfValues, payload);
	}

	public Gyro_RawDataPacket(DataPacket packet) throws Exception {
		super(ValidPacketTypes.gyro_raw_data_packet, numberOfValues, packet);
	}
	
	public XYZSensor GetAxisRotation() {
		AndroidSensor sensor = DeSerialize();
		return new XYZSensor(sensor.GetValues());
	}
}

File: Comms Protocol CSharp/Comms Protocol CSharp/Gyro_RawDataPacket.cs

namespace Comms_Protocol_CSharp
{
    class Gyro_RawDataPacket : AndroidDataPacket
    {
        private static int numberOfVals = 3;

        public Gyro_RawDataPacket() : base(ValidPacketTypes.gyro_raw_data_packet, numberOfVals)
        {

        }

        public Gyro_RawDataPacket(byte[] payload) : base(ValidPacketTypes.gyro_raw_data_packet, numberOfVals, payload)
        {

        }

        public Gyro_RawDataPacket(DataPacket packet) : base(ValidPacketTypes.gyro_raw_data_packet, numberOfVals, packet)
        {

        }

        public XYZSensor GetAxisRotation()
        {
            AndroidSensor sensor = DeSerialize();
            return new XYZSensor(sensor.GetValues());
        }
    }
}
